AWSAuthSDK/Sources/AWSUserPoolsSignIn/UserPoolsUI/AWSTableInputCell.m

```diff
@@ -36,7 +36,15 @@ - (IBAction)textEditingDidBegin:(id)sender {
         [self onTap];
     }
 }
-    
+
+- (void)layoutSubviews {
+    [super layoutSubviews];
+    // handle the case where the just added input has pre-set content
+    // so we can re-use the placeholder/label animation provided by onTap
+    if (![self.inputBox.text isEqual: @""] && self.placeHolderView.isHidden) {
+        [self onTap];
```

Comment:
1. This is probably not the right place to do this--`layoutSubviews` is typically reserved for layout calculations and constraint setup/reconciliation. Your comment says you are dealing with "just added input", so that suggests you want this in whatever setup is called from `dequeueReusableCell` or `cellForIndexPath`. (And then be sure to reset it in this class' `prepareForReuse` method)
1. rather than relying on `onTap` to always do what you want, extract the desired behavior into a separate helper method and invoke from both `layoutSubviews` and `onTap`.
1. Is laying out with animation the right thing to do in all cases? Initial layouts probably don't want to be animating individually; I'd expect us to rely on the table view to animate rows into place.

Reply:
For context: previously an overlay with a fixed background color was handling that case, after the color cleanup that hack stopped working.

1. I thought about this too, it didn't feel right but I could not find a place to run the placeholder/label logic as soon as the field is rendered, so in case it has content the proper visibility value is set for each component.
2. sounds good, I like that better too. There's already other calls to `onTap` and I'll also clean them up
3. the animation is just a show/hide the placeholder and/or label inside the cell, not the cell itself